File: app/src/main/java/com/wkz/viewpager/MainActivity.java
package com.wkz.viewpager;

import android.app.Dialog;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.hss01248.lib.MyItemDialogListener;
import com.hss01248.lib.StytledDialog;
import com.wkz.viewpager.transformer.AccordionTransformer;
import com.wkz.viewpager.transformer.BackgroundToForegroundTransformer;
import com.wkz.viewpager.transformer.CubeInTransformer;
import com.wkz.viewpager.transformer.CubeOutTransformer;
import com.wkz.viewpager.transformer.DefaultTransformer;
import com.wkz.viewpager.transformer.DepthPageTransformer;
import com.wkz.viewpager.transformer.FlipHorizontalTransformer;
import com.wkz.viewpager.transformer.FlipVerticalTransformer;
import com.wkz.viewpager.transformer.ForegroundToBackgroundTransformer;
import com.wkz.viewpager.transformer.OverspreadTransformer;
import com.wkz.viewpager.transformer.RotateDownTransformer;
import com.wkz.viewpager.transformer.RotateUpTransformer;
import com.wkz.viewpager.transformer.ScaleInOutTransformer;
import com.wkz.viewpager.transformer.StackTransformer;
import com.wkz.viewpager.transformer.TabletTransformer;
import com.wkz.viewpager.transformer.ZoomInTransformer;
import com.wkz.viewpager.transformer.ZoomOutSlideTransformer;
import com.wkz.viewpager.transformer.ZoomOutTransformer;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Locale;
import java.util.Map;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private ViewPager mVpPager;
    Dialog bottomSheetDialog;
    /**
     * 切换转换效果
     */
    private Button mBtnChangeTransformer;
    private Map<String, ViewPager.PageTransformer> mTransformerMap = new HashMap<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        mVpPager = (ViewPager) findViewById(R.id.vp_pager);
        mBtnChangeTransformer = (Button) findViewById(R.id.btn_change_transformer);
        mBtnChangeTransformer.setOnClickListener(this);

        mVpPager.setPageTransformer(true, new DefaultTransformer());
        mVpPager.setAdapter(new PagerAdapter() {
            @Override
            public Object instantiateItem(ViewGroup container, int position) {

                FrameLayout page = new FrameLayout(MainActivity.this);
                TextView mText = new TextView(MainActivity.this);
                int bg = Color.rgb((int) Math.floor(Math.random() * 128) + 64,
                        (int) Math.floor(Math.random() * 128) + 64,
                        (int) Math.floor(Math.random() * 128) + 64);
                mText.setBackgroundColor(bg);
                mText.setGravity(Gravity.CENTER);
                mText.setText(String.format(Locale.getDefault(), "Page %d", position + 1));
                mText.setTextSize(30);
                mText.setTextColor(Color.WHITE);
                mText.setTag(OverspreadTransformer.TAG_PARALLAX);

                page.addView(mText, FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT);

                container.addView(page, ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
                return page;
            }

            @Override
            public int getCount() {
                return 10;
            }

            @Override
            public boolean isViewFromObject(View view, Object object) {
                return view == object;
            }

            @Override
            public void destroyItem(ViewGroup container, int position, Object object) {
                container.removeView((View) object);
            }

            @Override
            public int getItemPosition(Object object) {
                return POSITION_NONE;
            }
        });


        //底部弹出的带或不带取消按钮的弹窗
        mTransformerMap.put("AccordionTransformer", new AccordionTransformer());
        mTransformerMap.put("BackgroundToForegroundTransformer", new BackgroundToForegroundTransformer());
        mTransformerMap.put("CubeInTransformer", new CubeInTransformer());
        mTransformerMap.put("CubeOutTransformer", new CubeOutTransformer());
        mTransformerMap.put("DefaultTransformer", new DefaultTransformer());
        mTransformerMap.put("DepthPageTransformer", new DepthPageTransformer());
        mTransformerMap.put("FlipHorizontalTransformer", new FlipHorizontalTransformer());
        mTransformerMap.put("FlipVerticalTransformer", new FlipVerticalTransformer());
        mTransformerMap.put("ForegroundToBackgroundTransformer", new ForegroundToBackgroundTransformer());
        mTransformerMap.put("OverspreadTransformer", new OverspreadTransformer());
        mTransformerMap.put("RotateDownTransformer", new RotateDownTransformer());
        mTransformerMap.put("RotateUpTransformer", new RotateUpTransformer());
        mTransformerMap.put("ScaleInOutTransformer", new ScaleInOutTransformer());
        mTransformerMap.put("StackTransformer", new StackTransformer());
        mTransformerMap.put("TabletTransformer", new TabletTransformer());
        mTransformerMap.put("ZoomInTransformer", new ZoomInTransformer());
        mTransformerMap.put("ZoomOutSlideTransformer", new ZoomOutSlideTransformer());
        mTransformerMap.put("ZoomOutTransformer", new ZoomOutTransformer());

        final List<String> strings = Arrays.asList(
                "AccordionTransformer",
                "BackgroundToForegroundTransformer",
                "CubeInTransformer",
                "CubeOutTransformer",
                "DefaultTransformer",
                "DepthPageTransformer",
                "FlipHorizontalTransformer",
                "FlipVerticalTransformer",
                "ForegroundToBackgroundTransformer",
                "OverspreadTransformer",
                "RotateDownTransformer",
                "RotateUpTransformer",
                "ScaleInOutTransformer",
                "StackTransformer",
                "TabletTransformer",
                "ZoomInTransformer",
                "ZoomOutSlideTransformer",
                "ZoomOutTransformer"
        );

        bottomSheetDialog = StytledDialog.showBottomItemDialog(MainActivity.this, strings, "cancel", true, true, new MyItemDialogListener() {
            @Override
            public void onItemClick(String text, int position) {
                Toast.makeText(MainActivity.this, text, Toast.LENGTH_SHORT).show();
                if (mTransformerMap.get(text) != null) {
                    mVpPager.setPageTransformer(true, mTransformerMap.get(text));
                }
            }

            @Override
            public void onBottomBtnClick() {
            }
        });

        bottomSheetDialog.dismiss();
    }

    @Override
    public void onBackPressed() {
        if (bottomSheetDialog != null && bottomSheetDialog.isShowing()) {
            bottomSheetDialog.dismiss();
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_change_transformer:
                bottomSheetDialog.show();
                break;
            default:
        }
    }
}

File: app/src/main/java/com/wkz/viewpager/MainActivity.kt
package com.wkz.viewpager

import android.app.Dialog
import android.graphics.Color
import android.os.Bundle
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.FrameLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import androidx.viewpager2.widget.ViewPager2
import com.fphoenixcorneae.transformer.*
import com.hss01248.lib.MyItemDialogListener
import com.hss01248.lib.StytledDialog
import java.util.*
import kotlin.math.floor

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var mVpPager: ViewPager2? = null
    var bottomSheetDialog: Dialog? = null

    /**
     * 切换转换效果
     */
    private var mBtnChangeTransformer: Button? = null
    private val mTransformerMap: MutableMap<String, ViewPager2.PageTransformer?> = HashMap()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
    }

    private fun initView() {
        mVpPager = findViewById<View>(R.id.vp_pager) as ViewPager2
        mBtnChangeTransformer = findViewById<View>(R.id.btn_change_transformer) as Button
        mBtnChangeTransformer!!.setOnClickListener(this)
        mVpPager!!.setPageTransformer(DefaultTransformer())
        mVpPager!!.adapter = object : RecyclerView.Adapter<MyViewHolder>() {
            override fun onCreateViewHolder(
                parent: ViewGroup,
                viewType: Int
            ): MyViewHolder {
                val page = FrameLayout(this@MainActivity).apply {
                    layoutParams = RecyclerView.LayoutParams(-1, -1)
                }
                return MyViewHolder(page)
            }

            override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
                holder.textView.text = String.format(Locale.getDefault(), "Page %d", position + 1)
            }

            override fun getItemCount(): Int {
                return 10
            }
        }


        //底部弹出的带或不带取消按钮的弹窗
        mTransformerMap["AccordionTransformer"] = AccordionTransformer()
        mTransformerMap["BackgroundToForegroundTransformer"] = BackgroundToForegroundTransformer()
        mTransformerMap["CubeInTransformer"] = CubeInTransformer()
        mTransformerMap["CubeOutTransformer"] = CubeOutTransformer()
        mTransformerMap["DefaultTransformer"] = DefaultTransformer()
        mTransformerMap["DepthPageTransformer"] = DepthPageTransformer()
        mTransformerMap["DrawerTransformer"] = DrawerTransformer()
        mTransformerMap["FadeOutFadeInTransformer"] = FadeOutFadeInTransformer()
        mTransformerMap["FlipHorizontalTransformer"] = FlipHorizontalTransformer()
        mTransformerMap["FlipVerticalTransformer"] = FlipVerticalTransformer()
        mTransformerMap["ForegroundToBackgroundTransformer"] = ForegroundToBackgroundTransformer()
        mTransformerMap["OverspreadTransformer"] = OverspreadTransformer()
        mTransformerMap["RotateDownTransformer"] = RotateDownTransformer()
        mTransformerMap["RotateUpTransformer"] = RotateUpTransformer()
        mTransformerMap["ScaleInOutTransformer"] = ScaleInOutTransformer()
        mTransformerMap["ScaleTransformer"] = ScaleTransformer()
        mTransformerMap["StackTransformer"] = StackTransformer()
        mTransformerMap["TabletTransformer"] = TabletTransformer()
        mTransformerMap["VerticalTransformer"] = VerticalTransformer()
        mTransformerMap["ZoomInTransformer"] = ZoomInTransformer()
        mTransformerMap["ZoomOutPageTransformer"] = ZoomOutPageTransformer()
        mTransformerMap["ZoomOutSlideTransformer"] = ZoomOutSlideTransformer()
        mTransformerMap["ZoomOutTransformer"] = ZoomOutTransformer()
        val strings = listOf(
            "AccordionTransformer",
            "BackgroundToForegroundTransformer",
            "CubeInTransformer",
            "CubeOutTransformer",
            "DefaultTransformer",
            "DepthPageTransformer",
            "DrawerTransformer",
            "FadeOutFadeInTransformer",
            "FlipHorizontalTransformer",
            "FlipVerticalTransformer",
            "ForegroundToBackgroundTransformer",
            "OverspreadTransformer",
            "RotateDownTransformer",
            "RotateUpTransformer",
            "ScaleInOutTransformer",
            "ScaleTransformer",
            "StackTransformer",
            "TabletTransformer",
            "VerticalTransformer",
            "ZoomInTransformer",
            "ZoomOutPageTransformer",
            "ZoomOutSlideTransformer",
            "ZoomOutTransformer"
        )
        bottomSheetDialog = StytledDialog.showBottomItemDialog(
            this@MainActivity,
            strings,
            "cancel",
            true,
            true,
            object : MyItemDialogListener() {
                override fun onItemClick(text: String, position: Int) {
                    Toast.makeText(this@MainActivity, text, Toast.LENGTH_SHORT).show()
                    if (mTransformerMap[text] != null) {
                        mVpPager!!.setPageTransformer(mTransformerMap[text])
                    }
                }

                override fun onBottomBtnClick() {}
            })
        bottomSheetDialog!!.dismiss()
    }

    internal class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var textView: TextView = TextView(view.context)

        init {
            val bg = Color.rgb(
                floor(Math.random() * 128).toInt() + 64,
                floor(Math.random() * 128).toInt() + 64,
                floor(Math.random() * 128).toInt() + 64
            )
            textView.setBackgroundColor(bg)
            textView.gravity = Gravity.CENTER
            textView.textSize = 30f
            textView.setTextColor(Color.WHITE)
            textView.tag = OverspreadTransformer.TAG_PARALLAX
            (view as FrameLayout).addView(textView)
            (textView.layoutParams as FrameLayout.LayoutParams).apply {
                width = FrameLayout.LayoutParams.MATCH_PARENT
                height = FrameLayout.LayoutParams.MATCH_PARENT
                gravity = Gravity.CENTER
            }
        }
    }

    override fun onBackPressed() {
        if (bottomSheetDialog != null && bottomSheetDialog!!.isShowing) {
            bottomSheetDialog!!.dismiss()
        } else {
            super.onBackPressed()
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_change_transformer -> bottomSheetDialog!!.show()
            else -> {
            }
        }
    }
}